prover/src/lib.rs

```diff
@@ -119,3 +123,77 @@ impl ProverInput {
         })
     }
 }
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct ProverInput {
+    pub blocks: Vec<BlockProverInput>,
+}
+
+impl ProverInput {
+    pub async fn prove(
+        self,
+        runtime: &Runtime,
+        checkpoint_proof: Option<GeneratedBlockProof>,
+        save_inputs_on_error: bool,
+    ) -> Result<Vec<GeneratedBlockProof>> {
+        let results: Arc<Mutex<HashMap<u64, Option<PlonkyProofIntern>>>> =
+            Arc::new(Mutex::new(HashMap::new()));
```

Comment:
Right, I was implementing this with #88 and #89 in mind. With the `BlockProofFuture` implemented, it will be very easy to parallelize proving task. Hence I would prefer not to downgrade this implementation despite it being synchronous.

Reply:
Resolving as non relevant anymore